src/components/ButtonV2/index.tsx

```diff
@@ -0,0 +1,238 @@
+              ? `1px solid ${theme.colors[variant].borderWeakHover}`
+              : 'none'
+          };
+        }
+
+        &:active {
+          outline : 3px solid #4F059940;
+        }
+      `
+          : /* Disabled */ `
+         
+          color : ${
+            variant === 'neutral'
+              ? theme.colors.neutral.textDisabled
+              : theme.colors[variant][colorValue]
+          };
+          border: ${
+            variant === 'neutral' || prominence === 'weak'
+              ? `1px solid ${
+                  variant === 'neutral'
+                    ? theme.colors.neutral.borderDisabled
+                    : theme.colors[variant].borderWeakDisabled
+                }`
+              : 'none'
+          };
+          `
+      }
+
+      ${
+        extend
+          ? `
+        transition: max-width 450ms ease;
+        max-width: 48px;
+        [data-text] { display: none; }
+
+        &:hover, &:focus {
+          [data-text] { display: inherit }
+          max-width: 275px;
+        }
+      `
+          : ''
+      }
+    `
+  }};
+`
+
+const FwdButton = ({
+  ariaLabel,
+  disabled = false,
+  className,
+  extend = false,
+  icon,
+  iconPosition = 'left',
+  innerRef,
+  isLoading = false,
+  onClick,
+  prominence = 'strong',
+  size = 'large',
+  target,
+  text,
+  to,
+  tooltip,
+  tooltipBaseId,
+  type = 'button',
+  variant = 'primary',
+}: ButtonProps) => {
+  const as = useMemo(() => {
+    if (to && !disabled) {
+      return UniversalLink as ElementType
+    }
+
+    return 'button'
+  }, [disabled, to])
+
+  const elementType = as === 'button' ? type : undefined
+
+  return (
+    <Tooltip id={tooltipBaseId} text={tooltip}>
+      <StyledButton
+        aria-disabled={disabled}
+        aria-label={ariaLabel || text}
+        as={as}
+        className={className}
+        ref={innerRef}
+        variant={variant}
+        size={size}
+        extend={extend}
+        prominence={variant === 'neutral' ? 'weak' : prominence}
+        disabled={disabled || isLoading}
+        onClick={!disabled && !isLoading ? onClick : undefined}
+        target={target}
+        to={to}
+        type={elementType}
+      >
+        {icon && !isLoading && iconPosition === 'left' ? (
+          <Icon name={icon} />
+        ) : null}
+        {text ? <span data-text>{text}</span> : null}
+        {icon && !isLoading && iconPosition === 'right' ? (
+          <Icon name={icon} />
+        ) : null}
+        {isLoading ? (
+          <Loader
+            active
+            trailColor="transparent"
+            color="currentColor"
+            size="1em"
+          />
+        ) : null}
+      </StyledButton>
+    </Tooltip>
+  )
+}
+
+const ButtonV2 = forwardRef<HTMLButtonElement, Omit<ButtonProps, 'innerRef'>>(
+  // eslint-disable-next-line react/jsx-props-no-spreading
+  (props, ref) => <FwdButton {...props} innerRef={ref} />,
+)
+
+ButtonV2.displayName = 'fwd(ButtonV2)'
```

Comment:
Args table of button is empty I think it might be because of that 🤔 

<img width="1053" alt="Screenshot 2022-06-28 at 09 49 07" src="https://user-images.githubusercontent.com/15812968/176123914-1320ec08-3053-4e12-b248-013c9da0c36c.png">

Reply:
Solve with a workaround ;)